### DISCUSSION TITLE
Setup debugging

### DISCUSSION THREAD

**Discussion by olivier1980** [dxx-rebirth/dxx-rebirth#761] [General]
I can build the project on Linux, but how can I configure CLion to run the game while debugging, ie using breakpoints etc?

Is there documentation on how the assets packs are structured, and what's in them? I'm trying to find a simple list of enemies, I could probably identify them by the `polygon_model_index` in struct `Robot_info`?

**Comment by vLKp** [comment]
Rebirth can be started from the terminal like any other graphical application.  If you intend to set breakpoints or allow a debugger to suspend the game, I strongly recommend that you include `-no-grab` in your command line options, so that the game never acquires a grab of your keyboard.  If the debugger suspends the game while the keyboard is grabbed, all keyboard input will be directed to a suspended process, which cannot release the grab until you unsuspend it.

---

Back when Descent was young, there was various community documentation about some of the file formats.  None of it was ever merged to the Rebirth repository, and I suspect much of it has been lost with the passage of time.  I believe some of it was hosted on a domain with "warp core" in the name.  Generally, your best option will be reading the game's code for loading the file format of interest.
